src/ape/managers/project.py

```diff
@@ -24,11 +25,98 @@
 )
 
 from .compilers import CompilerManager
-from .config import ConfigManager
+from .config import ConfigManager, GithubDependencyConfig
 from .converters import ConversionManager
+from .networks import NetworkManager
 
 
-class ProjectManager:
+class BaseProject:
```

Comment:
So the dependencies and the `ProjectManager` share some things, such as the cache file related logic.
The difference between the two is that the `ProjectManager` has access to the source files whereas the `Dependency` just downloads the source files in a temporary directory and builds the manifest file from that.
Everything related to that is in this file, there a few classes:

`BaseManager`
`ProjectManager`
`DependencyProject` - a base class that any dependency type should be able to use, such as `GitLab`, `IPFS`, etc.
`GithubDependency` - the main implementation we have the downloads or clones from github to build the manifest

Reply:
I am looking for feedback on the design of this.

I have considered we might want to init an `ape.api.projects` module where we can put the `BaseProject` and the `DependencyProject`.